### PR TITLE
[fix] clean up Go SDK Selectors doc

### PR DESCRIPTION
**Pull request by sw-yx**
we had some errant stuff slip through review...

also making sure to crosslink with other selectors references so that it is more discoverable

### CODE REVIEW

File: docs/go-execute-activity.md [hunk 1]
```diff
@@ -95,4 +95,4 @@ However, this is not necessary. If you want to execute multiple Activities in pa
 repeatedly call `workflow.ExecuteActivity()`, store the returned futures, and then wait for all
 Activities to complete by calling the `Get()` methods of the future at a later time.
 
-To implement more complex wait conditions on returned future objects, use the `workflow.Selector` class.
+To implement more complex wait conditions on returned future objects, use the `workflow.Selector` class (learn more on the [Go SDK Selectors](https://docs.temporal.io/docs/go-selectors) doc).
```
Comment (sw-yx) [inline docs/go-execute-activity.md:98]:
```suggestion
To implement more complex wait conditions on returned future objects, use the `workflow.Selector` class. Learn more on the [Go SDK Selectors](https://docs.temporal.io/docs/go-selectors) page.
```

File: docs/go-selectors.md [hunk 2]
```diff
@@ -39,13 +39,13 @@ func SampleWorkflow(ctx workflow.Context) error {
...
 		// matching on the channel doesn't consume the message.
-    // So it has to be explicitly consumed here
+	 	// So it has to be explicitly consumed here
 		c.Receive(ctx, &signalVal)
 		// do something with received information
 	})
 
 	// API Example: block until the next Future is ready to run
-  // important! none of the deferred code runs until you call selector.Select
+	// important! none of the deferred code runs until you call selector.Select
```
Comment (sw-yx) [inline docs/go-selectors.md:48]: cleaned up weird indents

File: docs/go-selectors.md [hunk 3]
```diff
@@ -101,21 +101,13 @@ Merely matching on the channel doesn't consume the message; it has to be explici
...
 Usage of Selectors is best learned by example:
 
-- Setting up a race condition between an Activity and a Timer, and conditionally execute ([example](https://github.com/temporalio/samples-go/blob/14980b3792cc3a8447318fefe9a73fe0a580d4b9/timer/workflow.go))
+- Setting up a race condition between an Activity and a Timer, and conditionally execute ([Timer example](https://github.com/temporalio/samples-go/blob/14980b3792cc3a8447318fefe9a73fe0a580d4b9/timer/workflow.go))
 - Receiving information in a Channel ([Mutex example](https://github.com/temporalio/samples-go/blob/14980b3792cc3a8447318fefe9a73fe0a580d4b9/mutex/mutex_workflow.go))
 - Looping through a list of work and scheduling them all in parallel ([DSL example](https://github.com/temporalio/samples-go/blob/14980b3792cc3a8447318fefe9a73fe0a580d4b9/dsl/workflow.go))
 - Executing activities in parallel, pick the first result, cancel remainder ([Pick First example](https://github.com/temporalio/samples-go/blob/14980b3792cc3a8447318fefe9a73fe0a580d4b9/pickfirst/pickfirst_workflow.go))
-
-Frequently Asked Questions:
-
```
Comment (sw-yx) [inline docs/go-selectors.md:116]: not meant for public, removed

File: sidebars.js [hunk 4]
```diff
@@ -91,7 +92,6 @@ module.exports = {
             'go-distributed-cron',
             'go-tracing',
             'go-search-apis',
-            'go-selectors',
```
Comment (sw-yx) [inline sidebars.js:94]: bumping up selectors in the sidebar to put them roughly nearer where they'd come up conceptually... what do you think?

or do we do this alphabetically at this point..

Comment (flossypurse) [reply sidebars.js:94]: I was thinking that eventually they will be ordered by level of most used -

**Review by flossypurse — APPROVED** [review]

**Pull request merged**